tests/ci/integration/run_strongswan_integration.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+
+# SYS_ROOT
+#  - SRC_ROOT(aws-lc)
+#    - SCRATCH_FOLDER
+#      - AWS_LC_BUILD_FOLDER
+#      - AWS_LC_INSTALL_FOLDER
+#      - STRONGSWAN_SRC_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER="${SRC_ROOT}/STRONGSWAN_BUILD_ROOT"
+STRONGSWAN_SRC_FOLDER="${SCRATCH_FOLDER}/strongswan"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+
+function strongswan_build() {
+  export CFLAGS="-I${AWS_LC_INSTALL_FOLDER}/include"
+  export LDFLAGS="-L${AWS_LC_INSTALL_FOLDER}/lib"
+  autoreconf -ivf
+  # Mirror the configuration flags set by strongSwan's CI by taking all of the
+  # flags specified within CONFIG when TEST is openssl-awslc in their test.sh.
+  # https://github.com/strongswan/strongswan/blob/44e241fccc166211ccfdd322047c1213ff3ae73c/scripts/test.sh#L468
+  ./configure --disable-defaults --enable-pki --enable-openssl --enable-pem \
+  --disable-dependency-tracking --enable-silent-rules --enable-test-vectors \
+  --enable-monolithic=no --enable-leak-detective=no --enable-asan
+  make -j ${NUM_CPU_THREADS}
+}
+
+function strongswan_run_tests() {
+  make -j ${NUM_CPU_THREADS} check
+}
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+git clone --depth 1 https://github.com/strongswan/strongswan.git ${STRONGSWAN_SRC_FOLDER}
```

Comment:
looks like we're testing against their mainline, do we expect them to frequently make potentially breaking changes? if so, we might want to pin this to a particular version. this isn't a concern if we get an AWS-LC integration test into StrongSwan's CI.

Reply:
I certainly don't expect them to do that, but maybe I'm naive? There will be an AWS-LC CI check added to strongSwan so perhaps we can be happy with a policy of mutually assured destruction?

Reply:
> There will be an AWS-LC CI check added to strongSwan

nice, I think this minimizes the chance of any breaking changes getting merged downstream.